drivers/pinctrl/pinctrl_hpmicro.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022 hpmicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#define DT_DRV_COMPAT hpmicro_hpm_pinctrl
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+#include <zephyr/drivers/pinctrl/pinctrl_hpmicro_common.h>
+
+#include <hpm_common.h>
+#include <hpm_soc.h>
+#define IOC_BASE_ADDRESS DT_REG_ADDR(DT_NODELABEL(pinctrl))
```

Comment:
This should be `DT_INST_REG_ADDR(0)`, not `DT_REG_ADDR(DT_NODELABEL(pinctrl))`

Reply:
> Why was this marked as resolved, without addressing the comment or stating why it shouldn't apply?

There's nothing wrong with writing it that way. Under the current module, other chips are also made in this way, if there is a problem, please change it uniformly.
[esp32](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_esp32.c)
[NXP](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_kinetis.c)
[GD](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_gd32_afio.c)